app/controllers/maintenance_requests_controller.rb

```diff
@@ -0,0 +1,62 @@
+class MaintenanceRequestsController < ApplicationController
+  helper_method :sort_column, :sort_direction
+
+  def index
+    @requests = MaintenanceRequest.joins(:user).joins(:route)
+    @requests = @requests.order(sort_column + " " + sort_direction)
+    @requests = @requests.not_resolved
+  end
+
+  def show
+    @request = MaintenanceRequest.joins(:user).joins(:route).find(params[:id])
```

Comment:
When you call `#user` or `#route` on `@request` for the first time, it queries the DB